src/commands/watch.rs

```diff
@@ -185,47 +180,63 @@ impl WatchCommand {
-                            return path.to_str();
+        let watch_patterns = app_manifest
+            .components
+            .iter()
+            .filter_map(|c| {
+                if let Some(build) = c.build.as_ref() {
+                    if build.watch.is_none() {
+                        // TODO: Make sure that documentation link is pointing specifically to spin watch docs
+                        println!("You haven't configured what to watch for the component: '{}'. Learn how to configure Spin watch at https://developer.fermyon.com",c.id);
+                    }
+                    build.watch.clone()
+                } else {
+                    // No build config for this component so lets watch the source instead
+                    if let RawModuleSource::FileReference(path) = &c.source {
+                        if let Some(string_path) = path.to_str() {
+                            return Some(vec![String::from(string_path)]);
                         }
-                        None
-                    })
-                    .map(String::from);
-                let component_file_patterns = app_manifest
-                    .components
-                    .iter()
-                    .filter_map(|c| c.wasm.files.as_ref())
-                    .flatten()
-                    .map(|raw_file_mount| match raw_file_mount {
-                        RawFileMount::Placement(raw_directory_placement) => String::from(
-                            raw_directory_placement
-                                .source
-                                .join("**/*")
-                                .to_str()
-                                .expect("conversion to str not to fail"),
-                        ),
-                        RawFileMount::Pattern(pattern) => pattern.to_string(),
-                    });
-                component_source_patterns
-                    .chain(component_file_patterns)
-                    .collect()
-            }
+                    }
+                    None
+                }
+            })
+            .flatten();
+        let component_source_patterns = app_manifest
+            .components
+            .iter()
+            .filter_map(|c| {
+                if let RawModuleSource::FileReference(path) = &c.source {
+                    return path.to_str();
+                }
+                None
+            })
+            .map(String::from);
+        let component_file_patterns = app_manifest
+            .components
+            .iter()
+            .filter_map(|c| c.wasm.files.as_ref())
+            .flatten()
+            .map(|raw_file_mount| match raw_file_mount {
+                RawFileMount::Placement(raw_directory_placement) => String::from(
+                    raw_directory_placement
+                        .source
+                        .join("**/*")
+                        .to_str()
+                        .expect("conversion to str not to fail"),
```

Comment:
I know this should never happen but I would feel a bit happier if this skipped or errored rather than panicking.  Can defer this though.